server/__tests__/integration/competition.test.js

```diff
@@ -0,0 +1,512 @@
+      expect(response.body.participants.map(m => m.username)).toContain('test player');
+
+      done();
+    });
+
+    test('Add participant to competition', async done => {
+      const response = await request.post(`/api/competitions/${TEST_DATA.minimal.id}/add`).send({
+        participants: ['new player'],
+        verificationCode: TEST_DATA.minimal.verificationCode
+      });
+
+      expect(response.status).toBe(200);
+      expect(response.body.newParticipants.length).toBe(1);
+      expect(response.body.newParticipants.map(m => m.username)).toContain('new player');
+
+      done();
+    });
+
+    test('Do not add participant to competition ( already a participant )', async done => {
+      const response = await request.post(`/api/competitions/${TEST_DATA.minimal.id}/add`).send({
+        participants: ['new player'],
+        verificationCode: TEST_DATA.minimal.verificationCode
+      });
+
+      expect(response.status).toBe(400);
+      expect(response.body.message).toBe('All players given are already competing.');
+
+      done();
+    });
+
+    test('Do not remove participant ( Competition not found )', async done => {
+      const response = await request.post(`/api/competitions/1234/remove`).send({
+        participants: ['new player'],
+        verificationCode: TEST_DATA.minimal.verificationCode
+      });
+
+      expect(response.status).toBe(404);
+      expect(response.body.message).toBe('Competition of id 1234 was not found.');
+
+      done();
+    });
+
+    test('Do not remove participant ( No participants givens )', async done => {
+      const response = await request.post(`/api/competitions/${TEST_DATA.minimal.id}/remove`).send({
+        participants: [],
+        verificationCode: TEST_DATA.minimal.verificationCode
+      });
+
+      expect(response.status).toBe(400);
+      expect(response.body.message).toBe('Invalid participants list.');
+
+      done();
+    });
+
+    test('Do not remove participant ( Participants not in list )', async done => {
+      const response = await request.post(`/api/competitions/${TEST_DATA.minimal.id}/remove`).send({
+        participants: ['epic player'],
+        verificationCode: TEST_DATA.minimal.verificationCode
+      });
+
+      expect(response.status).toBe(400);
+      expect(response.body.message).toBe('None of the players given were competing.');
+
+      done();
+    });
+
+    test('Remove participant', async done => {
+      const response = await request.post(`/api/competitions/${TEST_DATA.minimal.id}/remove`).send({
+        participants: ['new player'],
+        verificationCode: TEST_DATA.minimal.verificationCode
+      });
+
+      expect(response.status).toBe(200);
+      expect(response.body.message).toBe(
+        `Successfully removed 1 participants from competition of id: ${TEST_DATA.minimal.id}.`
+      );
+
+      done();
+    });
+
+    test('Do not remove participant ( Not in participant list )', async done => {
```

Comment:
Isn't this test repeated? line 447

Reply:
yup! seems like I was copy pasting a bit too much haha